src/Bicep.Decompiler/UniqueNamingResolver.cs

```diff
@@ -30,7 +30,7 @@ private static string GetNamingSuffix(NameType nameType)
 
         private static string EscapeIdentifier(string identifier)
         {
-            return Regex.Replace(identifier, "[^a-zA-Z0-9]+", "_").Trim('_');
+            return Regex.Replace(identifier, "[^a-zA-Z0-9\\p{Lu}\\p{Ll}\\p{Lt}\\p{Lm}\\p{Lo}\\p{Nl}\\p{Nd}\\p{Mn}\\p{Mc}\\p{Cf}]+", "_").Trim('_');
```

Comment:
>Lu [](start = 61, length = 2)

This is actually not the full set of unicode character classes. Pc is missing (which includes _). Adding the Pc class breaks a bunch of decompiler tests. There's also a preexisting issue #1360 that is related to this path as well.